### PR TITLE
luminous: mds: log evicted clients to clog/dbg

### PR DESCRIPTION
**Pull request by ashishkumsingh**
https://tracker.ceph.com/issues/37823

### CODE REVIEW

File: src/mds/MDSRank.cc [hunk 1]
```diff
@@ -3059,11 +3059,21 @@ bool MDSRank::evict_client(int64_t session_id,
     return false;
   }
 
+  auto& addr = session->info.inst.addr;
+  {
+    CachedStackStringStream _ss;
```
Comment (batrick) [inline src/mds/MDSRank.cc]: Please use `std::stringstream` instead.

**Comment by smithfarm** [comment]
@ashishkumsingh Failed make check, @batrick suggested a fix, please take a look.

**Comment by smithfarm** [comment]
Build failure is:

```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mds/MDSRank.cc: In member function ‘bool MDSRank::evict_client(int64_t, bool, bool, std::stringstream&, Context*)’:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mds/MDSRank.cc:3072:20: error: ‘std::stringstream {aka class std::__cxx11::basic_stringstream<char>}’ has no member named ‘get_stream’
     auto& ss = _ss.get_stream();
                    ^
src/mds/CMakeFiles/mds.dir/build.make:110: recipe for target 'src/mds/CMakeFiles/mds.dir/MDSRank.cc.o' failed
```

(Putting it here because the Jenkins console log will disappear within a very short time)

File: src/mds/MDSRank.cc [hunk 2]
```diff
@@ -3066,11 +3066,21 @@ bool MDSRank::evict_client(int64_t session_id,
...
   }
 
+  auto& addr = session->info.inst.addr;
+  {
+    std::stringstream _ss;
+    auto& ss = _ss.get_stream();
+    ss << "Evicting " << (blacklist ? "(and blacklisting) " : "")
+       << "client session " << session_id << " (" << addr << ")";
+    dout(1) << ss.strv() << dendl;
+    clog->info() << ss.strv();
```
Comment (smithfarm) [inline src/mds/MDSRank.cc]: I'm not well-versed in `std::stringstream`, but it doesn't look too complicated. Something like this:

```
{
  std::stringstream ss;
  ss << "Evicting " << (blacklist ? "(and blacklisting) " : "")
       << "client session " << session_id << " (" << addr << ")";
  dout(1) << ss.str() << dendl;
  clog->info() << ss.str();
}
```

@batrick Does that look right?

Comment (batrick) [reply src/mds/MDSRank.cc]: Yes!

Comment (ashishkumsingh) [reply src/mds/MDSRank.cc]: @smithfarm @batrick thank you for the help. I have made the changes and commit.

**Comment by yuriw** [comment]
[wip-yuri4-testing-2019-02-22-1534-luminous](https://trello.com/c/SqBDMx2n/546-wip-yuri4-testing-2019-02-22-1534-luminous)

**Pull request merged**